python/delta_sharing/rest_client.py

```diff
@@ -408,16 +409,28 @@ def list_files_in_table(
                 raise LookupError("Missing delta-table-version header")
 
             lines = values[1]
-            protocol_json = json.loads(next(lines))
-            metadata_json = json.loads(next(lines))
 
-            return ListFilesInTableResponse(
-                delta_table_version=int(headers.get(
-                    DataSharingRestClient.DELTA_TABLE_VERSION_HEADER)),
-                protocol=Protocol.from_json(protocol_json["protocol"]),
-                metadata=Metadata.from_json(metadata_json["metaData"]),
-                add_files=[AddFile.from_json(json.loads(file)["file"]) for file in lines],
-            )
+            if (DataSharingRestClient.CAPABILITIES_HEADER in headers and
+                    "responseformat=delta" in headers[DataSharingRestClient.CAPABILITIES_HEADER]):
```

Comment:
is there a better way than directly comparing the string "responseformat=delta"?

Reply:
the only other way I can think of is to try to serialize it into a json and try catch the serialization process. This might be extra overhead though